app/services/punches_service.rb

```diff
@@ -0,0 +1,15 @@
+class PunchesService
+  def self.create(punches, user)
+    deletes = punches.map { |punch| punch['from'] }
+
+    @punches = user.punches
```

Comment:
I don't think you need to use instance variables here, since you're not using them in the view layer.

Reply:
Cool, I didnt know this instance variables in ruby, I thought it was a special variable for transactions